### ISSUE TITLE
Can't start Dead Space with Durazno

### ISSUE THREAD

**Issue by janoi152** [KrossX/Durazno#21]
There is a known problem with very large dead zones in the PC version of Dead Space. I tried using Durazno to solve the issue but when I copy the xinput1_3.dll into the game folder, the game refuses to start. The game is 32bit so this issue happens with 32bit Durazno (with 64bit the game starts but ignores the Durazno settings). When I click the Play button in Steam client the game just changes to running for a few seconds and then changes to normaln (not running) again, without showing any sign of window or other activity. What could be the problem? There are other people who are using this same solution and it works for them.

**Comment by janoi152** [comment]
Oh and other thing I've noticed is that the controller test within the Durazno app isn't showing any activity from the controller when pressing buttons or using analogs (but the controller is working alright in any other apps/games). Thank you for your help.

**Comment by KrossX** [comment]
I know it used to work with Dead Space, though the test tab not working either makes it weird. Hmmm... try installing the DirectX runtimes and checking older versions of Durazno.  Just in case, this is with an X360 or XInput compatible controller right? 

Other than that, there's XInput Plus that seems to be better than Durazno overall. Might wanna give it a try.

**Comment by janoi152** [comment]
Thank you for the answer. I'm using DualShock 3 controller with SCP drivers, so it is masked in the system as xbox360 controller. But maybe the problem might be in the combination of Durazno wrapper with the SCP driver. Since I posted my issue I have tried the XInput Plus and it is working alright (even the within app testing reacts to the controller). So it really has something to do with the Durazno. But I would still rather use the Durazno because the XInput Plus doesn't have the same ability to counter the large dead zone in Dead Space (there is no AntiDeadZone functionality), I tried to max out the Linearity but that's still not quite right. I have DirectX properly installed so it shouldn't be a problem but I will try the older versions of Durazno.

**Comment by KrossX** [comment]
SCP drivers should have no issues with XInput indeed. I dunno what could be the problem at all, specially with the test tab on the tool not working.

Hmmm... is it the proper pad number? The green thingy on the top left of the tool lets you change between pad ports.

**Comment by janoi152** [comment]
Changing ports didn't work either, I tried that before. But I just tried version 0.6-27 and the test tab is working! Now I'm going to try it with the game.

**Comment by janoi152** [comment]
It's working great!! All I had to do was to use the older version 0.6-27. Thank you very much!

**Comment by KrossX** [comment]
Weird. It's possible you're missing the runtimes for latest one.
https://www.microsoft.com/en-us/download/details.aspx?id=48145

Missing runtimes would indeed make the dll loading fail, thus the test tab wouldn't work and the game could fail to load.

**Comment by janoi152** [comment]
Is there a way to tell which version I have now? I don't want to blindly install everything if it isn't really needed.

**Comment by KrossX** [comment]
It doesn't matter which ones you have, only matters that you have the appropriate ones. The 2015 runtimes are required since the dll was built with VisualStudio 2015. If the problem is missing runtimes, then you need those runtimes installed.

**Comment by janoi152** [comment]
Yes, now the test tab is working alright in the 0.6-29 version too. So the problem was really with the runtimes. Sorry for my early mistake I thought that I had the latest version of the Visual C++ runtimes.

**Comment by KrossX** [comment]
No problem. You need the specific runtimes depending on what was used to compile the binary. So a new application might require very old runtimes if that was the compiler used to make it.

Previous Durazno was built statically which avoids requiring extra runtimes, which is why the previous version worked fine. Latest change goes back to a default shared build requiring the usual runtimes. This with other changes were made to help certain games not loading Durazno properly.

**Comment by uladz** [comment]
Don't want to beat a dead horse but I can't get it working with Dead Space either. Test tab is working fine, I'm trying to adjust the anti-deadzone setting and it has zero effect in-game. Inspecting the running game process I can confirm that the Durazno xinput DLL is loaded (but a bunch of original XINPUT DLL are loaded as well at the same time). VC++ runtime libs are installed. I've tried the latest 0.6-43 and a few older including 0.6-29, none worked for me. Somehow Dead Space manages to hook up to the original xinput anyhow. Anything else I can try? Multiple sources say that it suppose to work with Dead Space. Thanks!

**Comment by KrossX** [comment]
Latest durazno comes with the DLLs already named and ready to be copied. Just copy them all (the 32bit version of them) where the main exe is and see if the ini file appears on first run. If it does, replace the ini file with your settings, if it doesn't, then Durazno won't work.

**Comment by vlad-unravel** [comment]
Thank you for the reply! The ini file is there (if I delete it it gets created when I start Dead Space) and I can confirm that the DLL gets loaded, at least I can see that that DLL is the list of loaded DLLs in Dead Space process list.

**Comment by KrossX** [comment]
Yet still the modifications don't apply. I have no idea then. Maybe there's something else hooking on XInput functions and bypasses wrappers.

**Comment by vlad-unravel** [comment]
Is there any way to debug it? Enable log, etc?

**Comment by KrossX** [comment]
Nope. Try with XInput Plus, as it's a better alternative overall.

**Comment by vlad-unravel** [comment]
I saw older version was writing out some sort of debug log. I took a quick look at the XInput Plus and it does not seem to support Anti-dadzone compensation, which is the only thing I need for Dead Space. And overall I like your tool better (except it does not work for me case :)). Again, multiple sources report that it suppose to work with Dead Space. It's Steam version of the game, I'm also running GeDoSaTo to enable 4K, Windows 10 Pro x64 if it matters. Thanks!
